src/Microsoft.AspNetCore.Razor.Tools/MetadataCache.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Razor.Tools
+{
+    internal class AssemblyReferenceCache
+    {
+        // Store 1000 entries -- arbitrary number
+        private const int CacheSize = 1000;
```

Comment:
In our primary scenarios, is 1000 big enough?

Reply:
I only tested this with `SimpleMvc` in our testapps. And the count for that is 147.